clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -82,6 +95,24 @@
 
     private final Time time;
 
+    private final Map<Integer, Collection<ApiVersionsResponse.ApiVersion>> nodeApiVersions = new HashMap<>();
+
+    private final Set<String> apiVersionsRequests = new HashSet<>();
```

Comment:
Would be nice to have `node` somewhere in this name. Maybe `nodesRequestingApiVersions`?